### DISCUSSION TITLE
Mixing the old style reducer with a new one triggers compiler errors I can't solve

### DISCUSSION THREAD

**Discussion by rurza** [pointfreeco/swift-composable-architecture#1708] [Q&A] [answered]
I started migrating my codebase to the new `ReducerProtocol`. The feature is ready and it compiles. I would like to "plug it" into the parent reducer alongside with sibling ad-hoc reducer that deals with the feature state. 
Unfortunately I'm getting compiler errors.
For some reason it expects result builder syntax and it's treating the whole "old style" reducer as the new one. I was trying to structure code differently, putting them on separate files, using different typealiases, using the "old" `Reducer` name – nothing helps. Any ideas?

<img width="1509" alt="Zrzut ekranu 2022-12-1 o 00 11 56" src="https://user-images.githubusercontent.com/2467137/204928366-30cad201-9ec6-43ca-9628-13285b3d8f6f.png">

**Comment by stephencelis** [comment, marked as answer]
@rurza Do you think you can whittle things down to a repro that we can troubleshoot? A screen shot unfortunately doesn't provide enough context. The error messages are also truncated, so even those full messages may help us troubleshoot a _little bit_ more (though a reduced repro of the error would be ideal).

**Reply by rurza** [reply]
Thank you Stephen! I was in the middle of working on an example when I discovered that my actions in the switch statement were wrong. I was specifying non existing cases and the compiler just lost his mind (despite colors in the editor showing that everything was fine)...
